### PR TITLE
Update go to 1.23

### PR DESCRIPTION
**Pull request by Ex4amp1e**
Issue: https://github.com/networkservicemesh/deployments-k8s/issues/12294

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -25,9 +25,9 @@ jobs:
       - name: Check out code
         uses: actions/checkout@v2
       - name: Setup Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v5
         with:
-          go-version: 1.20.5
+          go-version: 1.23.1
```
Comment (denis-tingaikin) [inline .github/workflows/ci.yaml:30]:
```suggestion
          go-version: 1.23.3
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -57,9 +57,9 @@ jobs:
     if: github.repository != 'networkservicemesh/cmd-template'
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-go@v1
+      - uses: actions/setup-go@v5
         with:
-          go-version: 1.20.5
+          go-version: 1.23.1
```
Comment (denis-tingaikin) [inline .github/workflows/ci.yaml:62]:
```suggestion
          go-version: 1.23.3
```

File: .golangci.yml [hunk 3]
```diff
@@ -147,25 +136,20 @@ linters:
...
-    - path: 'main.go'
-      linters:
-        - revive
-      text: 'package-comments: should have a package comment'
     - path: main.go
       linters:
         - funlen
-      text: "Function 'main'"
+        - gocritic
+        - revive
```
Comment (denis-tingaikin) [inline .golangci.yml:152]: Is it possible to disable just certain issues and keep these linters working with main.go?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Ex4amp1e) [reply .github/workflows/ci.yaml]: Done

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (Ex4amp1e) [reply .github/workflows/ci.yaml]: Done

File: .golangci.yml (hunk 3 above)
Comment (Ex4amp1e) [reply .golangci.yml]: Sure, investigated more, we had collision of gocritic and govet linter about shadowing 'err' variables and as i got -  this is convention to re-assign 'err', so applied gocritic fix and excluded govet 'err' shadowing

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**